src/libs/SelectionScraper/index.js

```diff
@@ -0,0 +1,115 @@
+import render from 'dom-serializer';
+import ExpensiMark from 'expensify-common/lib/ExpensiMark';
+import {parseDocument} from 'htmlparser2';
+import _ from 'underscore';
+
+const elementsWillBeSkipped = ['html', 'body'];
+const tagAttribute = 'data-testid';
+
+/**
+ * Reads html of selection. If there is no Selection API returns empty string.
+ * @returns {String} HTML of selection as String
+ */
+const getHTMLOfSelection = () => {
+    if (window.getSelection) {
+        const selection = window.getSelection();
+
+        if (selection.rangeCount > 0) {
+            const div = document.createElement('div');
+
+            // We traverse all ranges, and get closest node with data-testid and replace its contents with contents of
+            // range.
+            for (let i = 0; i < selection.rangeCount; i++) {
+                const range = selection.getRangeAt(i);
+
+                const clonedSelection = range.cloneContents();
+
+                // If clonedSelection has no text content this data has no meaning to us.
+                if (clonedSelection.textContent) {
+                    let node = null;
+
+                    if (range.commonAncestorContainer instanceof HTMLElement) {
+                        node = range.commonAncestorContainer.closest(`[${tagAttribute}]`);
+                    } else {
+                        node = range.commonAncestorContainer.parentNode.closest(`[${tagAttribute}]`);
+                    }
+
+                    // This means "range.commonAncestorContainer" is a text node. We simply get its parent node.
+                    if (!node) { node = range.commonAncestorContainer.parentNode; }
+
+                    node = node.cloneNode();
+                    node.appendChild(clonedSelection);
+                    div.appendChild(node);
+
+                    // We should not add new line after last range.
+                    if (i < selection.rangeCount - 1) {
+                        div.appendChild(document.createElement('br'));
+                    }
+                }
+            }
+
+            return div.innerHTML;
+        }
+
+        return window.getSelection().toString();
+    }
+
+    // If no Selection API returns empty string.
+    return '';
```

Comment:
Same comment as above, If no selection, API returns empty string - or maybe I'm just reading it incorrectly